orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -40,7 +42,9 @@ def _get_device_index(device):
                               inputs[idx].data_ptr())
 
     for value_info in model.graph.output:
-        io_binding.bind_output(value_info.name, device if device.find(':') == -1 else device[:device.find(':')])
+        io_binding.bind_output(value_info.name, device if device.find(':') == -1 else device[:device.find(':')],
```

Comment:
Maybe don't convert `self._device` to string when call `_create_iobinding`?
Then you can check `device.type` here instead of string search for ':'

Reply:
I'm trying to minimize the amount of refactoring in this change, but I do agree that the device handling seems like it could be simplified. I'll let @thiagocrepaldi take that on in a follow up, since he's best placed to know if there are any subtleties here.